staking/programs/integrity-pool/src/state/pool.rs

```diff
@@ -284,8 +284,7 @@ impl PoolData {
         if *publisher == Pubkey::default() {
             return Err(IntegrityPoolError::ThisCodeShouldBeUnreachable.into());
         }
-        publisher_caps
-            .caps
+        publisher_caps.caps()[..publisher_caps.num_publishers() as usize]
```

Comment:
I think we can change the `caps()` function to return only the valid caps instead of filtering it here.

Reply:
ty